packages/api/tests/rich-text-detection.test.ts

```diff
@@ -208,6 +209,57 @@ describe('detectFacets', () => {
       expect(Array.from(rt.segments(), segmentToOutput)).toEqual(outputs[i])
     }
   })
+
+  it('correctly detects tags inline', async () => {
+    const inputs: [string, string[]][] = [
+      ['#a', ['#a']],
+      ['#1', []],
+      ['#tag', ['#tag']],
+      ['body #tag', ['#tag']],
+      ['#tag body', ['#tag']],
+      ['body #tag body', ['#tag']],
+      ['body #1', []],
+      ['body #a1', ['#a1']],
+      ['#', []],
+      ['text #', []],
+      ['text # text', []],
+      [
+        'body #thisisa64characterstring_aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+        ['#thisisa64characterstring_aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'],
+      ],
+      [
+        'body #thisisa65characterstring_aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab',
+        [],
+      ],
+      ['its a #double#rainbow', ['#double#rainbow']],
+      ['some #n0n3s@n5e!', ['#n0n3s@n5e']],
+      ['works #with,punctuation', ['#with,punctuation']],
+      [
+        'strips trailing #punctuation, #like, #this.',
+        ['#punctuation', '#like', '#this'],
+      ],
+      ['strips #multi_trailing___...', ['#multi_trailing']],
```

Comment:
We don't want to detect trailing punctuation e.g. `I love #turtles!` should match `#turtles`. However, we should also strip multiple ending punctuation marks, like `I love #turtles!!!!!`.